### PR TITLE
Update Geth to 1.9.0

### PR DESCRIPTION
**Pull request by rmulhol**
This diff looks unnecessarily large because I deleted `vendor/` 🎉 🚀 ✂️ 

Would appreciate if folks would give a go at checking out the branch and running `make build`. The reason I deleted `vendor/` is that I was having trouble executing `make build` after running `go mod vendor` due to `.h` and `.c` files not getting pulled in. If it works with versioned deps specified in `go.mod` and no `vendor/`, I'm thinking we might be able to safely delete ~2 million lines of code.

Important: these changes require go 1.12 and `export GO111MODULE=on`

### CODE REVIEW

File: pkg/contract_watcher/shared/parser/parser_test.go [hunk 1]
```diff
@@ -220,7 +220,7 @@ var _ = Describe("Parser", func() {
 			Expect(err).ToNot(HaveOccurred())
 
 			selectMethods := p.GetMethods([]string{})
-			Expect(len(selectMethods)).To(Equal(22))
+			Expect(len(selectMethods)).To(Equal(25))
```
Comment (rmulhol) [inline pkg/contract_watcher/shared/parser/parser_test.go]: @i-norden it seems like the changes to Geth caused this function to return more methods - do you know if this is a problem?

Comment (i-norden) [reply pkg/contract_watcher/shared/parser/parser_test.go]: It seems they fixed something.. inspecting the ABI for that contract shows that there are 25 functions, not 22... Not sure what exactly was changed such that 3 of the functions that were previously not considered methods are now considered as such, but I don't think it is a problem either way. Will update if I find anything else out!

Comment (i-norden) [reply pkg/contract_watcher/shared/parser/parser_test.go]: Just noticed this is a WIP, I may have approved prematurely 🙃

**Comment by i-norden** [comment]
I wasn't able to figure out how to use un-versioned dependencies with `go mod` (I'm not sure it is possible) which caused some major issues with the seed node, state-diffing, and pg-ipfs work...

**Comment by rmulhol** [comment]
@i-norden with this approach the versions are specified in [`go.mod`](https://github.com/vulcanize/vulcanizedb/blob/a2b26346780ce4d1314fb123bbfead55a3910944/go.mod) - think that would resolve the issues you were running into?

**Comment by m0ar** [comment]
@i-norden Or do you mean you _need_ dependencies without version? :thinking:

**Comment by m0ar** [comment]
Also this is really sexy :100: :1st_place_medal:

**Comment by i-norden** [comment]
Yeah I mean dependencies that aren't versioned. `go-mod` restricts us to working with only versioned dependencies, so we can't use any forked repos or any work that hasn't been in an official release (afaik). Obviously it is best practice to only use versioned dependencies, but the reality is if we want to patch something together with, for example, our own fork of Geth or IPFS (i.e. the seed node work) I don't think we can do that with `go-mod`.

**Comment by rmulhol** [comment]
@i-norden why not just release a version of our fork?

**Comment by i-norden** [comment]
@rmulhol that's a good point, I'll just do that. I wasn't sure how version releases work on forks- those releases are only local to the fork and require no approval from the original repo maintainers? I see it being somewhat cumbersome having to do a new release of a dependency any time we want to test any changes we make to it- no matter how small- but I'll adjust!

**Comment by i-norden** [comment]
If the releases on a fork are local to that fork only, can `go-mod` find it and distinguish it from a release of the same version on the master branch?

**Comment by rmulhol** [comment]
@i-norden good question! I think this [replace directive](https://github.com/golang/go/wiki/Modules#when-should-i-use-the-replace-directive) might greatly simplify things - looks like you can even point to the local filesystem rather than creating a release. That said, I'm totally open to postponing the use of `go mod` if we want to test it out a bit first.

**Comment by i-norden** [comment]
@rmulhol awesome! That replace directive is exactly the thing I need- thanks for pointing that out, and sorry for missing that! Don't postpone for me, I think all my concerns have been allayed.

**Comment by elizabethengelman** [comment]
I'm able to build this with the go modules! 

However, I'm seeing a weird error when I run `composeAndExecute` on this branch (pointing to [this](https://github.com/vulcanize/mcd_transformers/blob/staging/environments/composeAndExecuteTransformers.toml) config file).

It looks to be trying to find an old configuration variable, but we don't have any references of that var in our current `mcd_transformers` code base on staging. 🤔 I'll share the stacktrace in slack to avoid sharing `mcd` specific code here.

**Comment by aaizuss** [comment]
I can build this with go modules 😄

**Pull request merged**